### ISSUE TITLE
Amplify codegen error creating graphql queries.ts/API.ts files in TypeScript

### ISSUE THREAD

**Issue by ilfoxo** [aws-amplify/amplify-codegen#736]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v18.17.0

### Amplify CLI Version

12.5.2

### What operating system are you using?

Windows

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Describe the bug

When I define a query in the schema.graphql without variables there is a error generating the **queries.ts** and **API.ts**.

For example, if the definition of the schema.graphql is like this:

```
type Query {
	generarCatalogoExcel: String
}
```

The definition created by amplify in **queries.ts**:
```
export const generarCatalogoExcel = /* GraphQL */ `query GenerarCatalogoExcel {
  generarCatalogoExcel
}
` as GeneratedQuery<
  APITypes.GenerarCatalogoExcelQueryVariables,
  APITypes.GenerarCatalogoExcelQuery
>;
```

And the **API.ts**
```
export type GenerarCatalogoExcelQuery = {
  generarCatalogoExcel?: string | null,
};
```

queries.ts is trying to get the definition _APITypes.GenerarCatalogoExcelQueryVariables_ from API.ts but it is not defined.

My workaround for now is to define GenerarCatalogoExcelQueryVariables in API.ts like this:
```
export type GenerarCatalogoExcelQueryVariables = {}
export type GenerarCatalogoExcelQuery = {
  generarCatalogoExcel?: string | null,
};
```

I've seen the same behavior with queries.ts and subscriptions.ts, I haven't tried mutations.ts

### Expected behavior

The definitions of Variables should be generated in API.ts despite I user variables or not in schema.graphql

### Reproduction steps

1. Define a query in schema.graphql without variables:
```
type Query {
	generarCatalogoExcel: String
}
```
2. run amplify codegen
3. check queries.ts and API.ts

### Project Identifier

Project Identifier: ce2327c5a7cee473d9214875c372c40f

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by alharris-at** [comment]
Confirmed this issue. Looking into a fix.

**Comment by alharris-at** [comment]
Confirming this has been resolved as of at least amplify CLI v12.8.2

Given the input schema:
```graphql
# schema.graphql

type Query {
  generarCatalogoExcel: String
}
```

And with typescript codegen configured I see the following generated graphql and types.

```typescript
// src/API.ts

export type GenerarCatalogoExcelQueryVariables = {
};

export type GenerarCatalogoExcelQuery = {
  generarCatalogoExcel?: string | null,
};
```

```typescript
// src/graphql/queries.ts

import * as APITypes from "../API";
type GeneratedQuery<InputType, OutputType> = string & {
  __generatedQueryInput: InputType;
  __generatedQueryOutput: OutputType;
};

export const generarCatalogoExcel = /* GraphQL */ `query GenerarCatalogoExcel {
  generarCatalogoExcel
}
` as GeneratedQuery<
  APITypes.GenerarCatalogoExcelQueryVariables,
  APITypes.GenerarCatalogoExcelQuery
>;

```
